docs/source/common/weights_loading.rst

```diff
@@ -216,3 +179,180 @@ do the following:
+
+|
+
+    .. testcode::
+
+        from pytorch_lightning.callbacks import ModelCheckpoint
+
+
+        class LitAutoEncoder(LightningModule):
+            def validation_step(self, batch, batch_idx):
+                x, y = batch
+                y_hat = self.backbone(x)
+
+                # 1. calculate loss
+                loss = F.cross_entropy(y_hat, y)
+
+                # 2. log val_loss
+                self.log("val_loss", loss)
+
+
+        # 3. Init ModelCheckpoint callback, monitoring "val_loss"
+        checkpoint_callback = ModelCheckpoint(monitor="val_loss")
+
+        # 4. Add your callback to the callbacks list
+        trainer = Trainer(callbacks=[checkpoint_callback])
+
+- It gives you the ability to specify the ``dirpath`` and ``filename`` for your checkpoints. Filename can also be dynamic so you can inject the metrics that are being logged using
+  :meth:`~pytorch_lightning.core.lightning.LightningModule.log`.
+
+|
+
+    .. testcode::
+
+        from pytorch_lightning.callbacks import ModelCheckpoint
+
+
+        # saves a file like: my/path/sample-mnist-epoch=02-val_loss=0.32.ckpt
+        checkpoint_callback = ModelCheckpoint(
+            dirpath="my/path/",
+            filename="sample-mnist-{epoch:02d}-{val_loss:.2f}",
+        )
+
+- You can save top-K and last-K checkpoints by configuring the ``monitor`` and ``save_top_k`` argument.
+
+|
+
+    .. testcode::
+
+        from pytorch_lightning.callbacks import ModelCheckpoint
+
+
+        # saves top-K checkpoints based on "val_loss" metric
+        checkpoint_callback = ModelCheckpoint(
+            monitor="val_loss",
+            mode="min",
+            dirpath="my/path/",
+            filename="sample-mnist-{epoch:02d}-{val_loss:.2f}",
+        )
+
+        # saves last-K checkpoints based on "global_step" metric
+        # make sure you log it inside your LightningModule
+        checkpoint_callback = ModelCheckpoint(
+            monitor="global_step",
+            mode="max",
+            dirpath="my/path/",
+            filename="sample-mnist-{epoch:02d}-{global_step}",
+        )
+
+- When using iterative training which doesn't have an epoch, you can checkpoint at every ``N`` training steps by specifying ``every_n_training_steps=N``.
+- In case your training is interrupted for some reason, you can specify ``save_last=True`` to ensure that a checkpoint is being created at the end so that you can
+  resume it or use it for :doc:`fault-tolerant training <../advanced/fault_tolerant_training>`.
+- You can also control the interval of epochs between checkpoints using ``every_n_epochs`` between checkpoints, to avoid slowdowns.
+- By default ``ModelCheckpoint`` callback saves model weights, optimizer states, etc., but in case you have limited disk space or just need the model weights to be saved you can specify ``save_weights_only=True``.
```

Comment:
```suggestion
- By default, the``ModelCheckpoint`` callback saves model weights, optimizer states, etc., but in case you have limited disk space or just need the model weights to be saved you can specify ``save_weights_only=True``.
```